test/browser/render.test.js

```diff
@@ -1379,6 +1379,74 @@ describe('render()', () => {
 			newScratch.parentNode.removeChild(newScratch);
 		});
 
+		it('should unmount existing components in prerendered HTML', () => {
+			const newScratch = setupScratch();
+			const unmount = sinon.spy();
+			const mount = sinon.spy();
+			class App extends Component {
+				componentDidMount() {
+					mount();
+				}
+
+				componentWillUnmount() {
+					unmount();
+				}
+
+				render() {
+					return <span>App</span>;
+				}
+			}
+
+			newScratch.innerHTML = `<div id="child"></div>`;
+
+			const childContainer = newScratch.querySelector('#child');
+
+			render(<App />, childContainer);
+			expect(childContainer.innerHTML.trim()).to.equal('<span>App</span>');
+			expect(mount).to.be.calledOnce;
+			render(<div />, newScratch, newScratch.firstElementChild);
+			expect(newScratch.innerHTML.trim()).to.equal('<div id=""></div>');
+			expect(unmount).to.be.calledOnce;
+
+			newScratch.parentNode.removeChild(newScratch);
+		});
+
+		it('should unmount nested existing components in prerendered HTML', () => {
+			const newScratch = setupScratch();
+			const unmount = sinon.spy();
+			const mount = sinon.spy();
+			class App extends Component {
+				componentDidMount() {
+					mount();
+				}
+
+				componentWillUnmount() {
+					unmount();
+				}
+
+				render() {
+					return <span>App</span>;
+				}
+			}
+
+			// The only difference to the previous test is the extra div here
```

Comment:
Yes but this `<div>` is not part of your vnode tree since you start underneath and then target it with `newScratch.firstElementChild` did this work with Preact 8? This isn't supported in Preact X atleast since this falls outside your render root